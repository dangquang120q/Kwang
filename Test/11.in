38
29
848852422 222410501 114826526 728159492 766285221 161534751 310268857 197986017 815150031 635655137 567615636 379203693 579019800 905881042 588326009 829067194 545788766 349420754 664484773 182239672 902205017 281838387 360114813 159683585 293294469 753504716 179774385 119278828 128444333 
15
569912100 390387832 900541422 628253992 367827066 325670956 363637277 302229037 649276408 641913843 336035550 347817586 211009650 722629327 19840118 
64
666315032 880104995 409285296 993930378 11100108 343005581 451298680 541282446 258282467 787997585 53732769 18393129 486466257 32362323 719455767 623102660 853236671 298239924 231761517 227460037 732959237 16470192 277163934 818490451 930937503 13179929 177580273 203856548 826866559 460564110 856661138 301388491 345799081 785828763 306263887 789070645 262743070 85154573 457685485 517660835 956882807 64201983 587057528 567865631 125141812 278595013 517728896 443953905 319960135 414141249 132021114 19646896 967142192 332192667 343146948 495534453 487127498 763458587 210210278 614703988 956620267 34521920 803964743 568776490 
76
14087450 229051790 211000646 587683178 516181162 50918400 73189064 231671726 971863502 947222628 175714710 854235505 741712239 317088078 271350358 677220855 71771421 44367437 256089109 954914456 597420041 148302023 434939347 965188674 403318486 208551979 585080943 931979606 500873582 289614332 444820045 881153376 255150752 818842231 801585538 837952673 182673524 879278703 963125055 348472235 382547342 562830973 623921672 41115587 651259162 329033625 961039925 217591680 908186593 190597497 950701131 427882997 234078195 840028045 951925101 520471916 114377498 440710068 429050519 868677436 750457094 684983208 161412312 858298248 546056035 869234627 759620068 913620217 540354271 690755718 945120895 727349358 73171555 31173643 54775606 97967036 
97
691885285 601417448 809991323 308991252 128218165 378882546 428308641 605129575 997931154 201734748 407627766 346911230 421993550 150296870 762655557 889916039 989544996 321770298 101238871 204990481 483586172 848752584 590364210 230882370 312328995 993156772 150908885 308348750 210257525 648336316 308105561 927133400 228168890 675223991 163223152 513480172 89916123 792237847 931570935 149665573 772977833 379384466 611090798 963414559 997767510 59394261 390849093 891131483 435025995 144360446 578158900 985958298 740304701 319961197 476620377 373902530 662475517 417887540 851333257 516104204 71980541 980510211 187164660 644816154 847110034 240466893 642806281 78718681 412545924 702448876 734486605 71007740 123127716 597788695 438260108 229879413 269602674 915245791 265256662 54218178 391404747 242683277 98402721 713858822 315475750 675025543 264548073 56248445 370379058 770567837 914070585 273457716 208690156 988802836 50248555 27348102 986119225 
21
866991762 345648000 259948700 329433327 961695814 640267404 859491137 356522595 26199315 841201999 419196968 57719986 536711890 714890821 470746554 17869617 962569712 232510429 452851192 702903740 71087481 
99
859027297 986907153 120005117 85997054 754809242 117503526 661504145 489157562 588032280 830930721 606047112 356588872 217483771 134817503 336075332 649896641 253386564 217014291 446818718 129000992 178234636 440642838 682149260 913111925 270309788 959085887 683989528 59277251 52649958 254749306 520137006 434972948 750681061 728946663 910086180 547834979 534941786 503370381 524600451 84735861 676854016 662191499 219661519 945401654 639285442 964472656 211847230 16888606 535804313 583700218 97052708 881687199 252413509 715918780 865936514 858125752 769769976 350503512 226983902 797335145 302156411 146819681 698657506 186540218 524132069 193327959 388764249 950006971 377245190 470374836 87427262 859953715 113087590 472355720 608439964 304924541 747448064 625267946 519243644 742476225 135708233 333790246 365613662 368067644 311133572 498826815 163205516 244331415 764111427 445802917 166326963 321935656 503404399 18820138 187492030 317482169 716449732 721293717 44571940 
34
485439675 59809745 713065810 147182757 492347703 303072125 857709085 370232419 875575287 991769437 665718857 363410136 511242459 218356503 87536671 340523956 298638093 488226334 807157923 849260506 184427945 258738136 885896614 705247004 216456961 866346919 506692832 42741781 827616384 981755072 599991686 385095900 282084254 884847291 
85
84336080 295586268 882027271 22752033 295204048 409551404 263953835 637815177 616640557 952566082 53638271 18177852 129723498 37070753 297863455 14732339 187343545 851262375 787322088 77220526 658121805 392816763 787705364 156925727 428985996 643482689 82018185 365144729 18593767 285209800 927230084 507086041 514620785 84409858 481448761 647715742 612090010 237641723 263560495 360400192 241551688 260295385 693573426 187877460 54053677 242212144 522383524 467555634 799498835 521605088 388482754 86545252 465770755 923924104 371191147 974944450 838404509 986569794 59953885 190457484 473917586 475544679 71983032 833288450 218641653 584450359 660230012 83134598 278816852 69516225 27231459 736370921 333842255 392819124 152886616 90113667 936238817 745274407 617546003 38251605 226800859 733915011 269704038 247713031 832611489 
34
82937266 52391884 337676295 870076814 953619526 144207271 60359449 968488735 625783364 213591054 271568264 304709327 800063032 567819358 991098455 305233323 948890502 450666304 606134007 382719986 490711822 99743923 164265403 132432976 655477807 573416090 421309776 645579868 574272466 834067548 46887876 661562905 498595300 708695419 
44
826935383 696292070 122835232 379250184 627183665 63297981 876082264 279625970 873322604 246976535 947646972 552450209 38550017 834080778 441960350 747582786 171647089 155025431 563217769 823084003 673037123 906005218 339368867 331070244 134662671 119968407 85256579 669142797 208912553 543009006 530474895 532712653 969810090 381243498 44681998 815160625 666540369 954778156 311119152 922912431 114435273 823449369 492280472 432748456 
58
474456011 661850084 154794086 812540789 320841801 277838693 570337538 42571263 505239108 869654828 499912975 489389067 595686880 319758192 613397345 94976288 183591836 924233587 265870234 958958132 585375832 96290297 770560791 133199010 90232470 617562531 554446275 995689803 960625526 796172693 708813893 399390344 51081396 109056952 148287105 324015659 127496166 989909016 906000608 870619609 410822597 712731174 561484490 416732762 811314380 640534203 718397807 180150123 503039840 779543452 739386285 978507383 316224745 285097846 735841942 277642800 948069643 643334402 
98
479575318 842995967 392312628 976744851 638628175 147161964 55506797 60761004 72112316 451539978 235323443 661323996 14666261 548564975 406439877 561551380 520564912 538015292 307398764 852851197 65123805 539408571 482312918 384647029 81203182 912440358 596400229 142135695 453350051 410886514 552800643 213423195 192933112 228472039 690767728 694966003 333776435 271459528 861536213 164159473 505182382 884282599 265009080 536008972 165248711 329127352 39762235 590898723 261465475 325674994 111629883 110761565 987528827 971457163 191475606 812897850 193427408 219220509 253872539 505196810 984518230 83886228 702452621 784694788 191423804 454001595 437337240 826171549 816143740 52534632 790823565 195972290 989989323 967047101 28708049 140390255 145700165 226721973 253554849 878790878 195902506 300337730 787879368 203445666 622878408 672950176 99609361 242055827 152657037 757427992 313770797 525098786 756024335 599591858 786983313 510343365 580681403 328860316 
31
55844195 395875799 372139823 539489636 703677714 197656446 486729625 282646176 131012427 445440080 420363713 339402171 611004574 840256022 942832634 337433943 981676022 298769293 983434067 184546654 13554120 666294802 937864659 164125614 171220340 535637171 221023366 200913373 181870720 482965195 399746783 
12
491805680 993663389 951878383 281061618 478940617 202729058 111565688 100522534 715058717 675097745 986155431 780030745 
40
361019960 66686372 399131799 192833252 529965198 282599385 329306252 923314162 247294307 116954173 646877997 630085185 451562984 126612343 67921145 130143234 282691075 75948024 377279974 260790367 551041933 773057852 283573874 424545532 968544591 700009303 21395603 754673107 343094634 338303786 806764374 13691049 522117159 367516819 927294142 601415341 502787954 936376488 205223229 621236542 
48
463373328 646900901 134114482 142912455 97593538 164312412 925268187 890513459 172592388 577907756 603169820 159104860 605855837 978607615 944344352 991660434 306323400 559892835 312734011 759207989 627247765 600541022 175372966 709821480 869846942 905124936 552997691 643178736 317634918 647078011 30004008 480210428 204262579 109986974 248326818 849615515 671328982 187051155 316737099 643876669 750908958 113376620 718939067 454614845 937802240 505290667 641770278 916413201 
17
122981486 376086241 804960231 983231737 936255488 287563965 219701012 687882586 408503933 27945846 583623063 207341283 579462837 48149309 264944686 633519012 691962518 
36
912468397 321481554 157287611 452196015 842733586 754397315 235254976 150403517 319723944 725577693 243617094 338247620 681163616 721354639 903203328 512757708 664174139 239523229 371587503 979882019 355443151 657344828 561420811 509742212 425905824 780768256 591849754 138521272 130113562 550096407 706238920 847709753 436602402 207306694 769560502 579568273 
38
458508017 819382184 488007518 257089505 903430380 170975675 524651629 451851508 312910816 773482902 159079876 610234992 365389419 393389568 379334329 782978030 46773411 139746790 651354446 102243978 825552461 244348447 732237380 450247264 251765569 411837411 285659899 497082891 555069212 756768608 80458680 954219437 281316041 404960468 802095083 261740351 716353487 417205255 
84
730484991 353705429 380756241 80039258 257616323 862713264 490040811 787695424 461992349 529709699 558827645 916469915 510781933 40425871 606726232 409806709 177871072 197476677 262870450 642024961 526107357 64284892 810000516 650755992 820145048 568269580 507387121 345781139 146354117 610470611 760586263 276783110 325013560 49444598 987815195 147027240 984298169 116666775 153483395 608968818 60913151 424677341 572923791 61368619 506526826 541091702 243121932 932137825 767045671 779758769 875374056 370971045 950436318 227024076 282648608 20427235 402582241 96720949 62345797 631446832 844187610 685387074 119039359 250138364 259661453 392274941 282117280 39598665 104700614 984770774 689934984 327431498 874414224 614009094 726349079 400631154 979475240 362434716 431121707 827419970 993842582 423172298 846517808 165092464 
87
793007586 447742125 780211892 875825496 942975213 682941701 804042327 578777471 163465587 365303897 873303847 795556687 260975653 140634830 154446874 542105624 683714405 475335388 104193748 749652468 469497040 386140125 530950856 107299656 37376868 198764739 492318792 680713068 87419416 684713480 535419843 295574513 608788342 839004375 776519823 655813074 701589621 976123437 182487450 909128252 404849454 603652189 207408805 652980270 235642989 842610391 386379719 275190929 809994443 270911914 882908421 874180670 44896992 121067995 594599735 182555014 634633108 449513332 266825581 172630468 220264883 538252818 358045319 919965877 386291478 243573427 564681236 324658139 345739391 73737608 467136503 574717862 465277356 139656665 321284205 198575082 788662224 699204294 247331634 883749687 839240503 363408644 87697855 96092526 441217511 40952625 197456598 
80
64716107 509866912 330584292 813907218 535207763 23197327 685533001 286420225 459541754 237553981 347561951 184350291 766059137 297092349 585419802 705923037 537931490 753496489 312192439 74018603 559420972 704435673 356665827 129503145 954578260 216830032 156811775 529163901 323735542 569864536 792533863 177733644 817154734 782531104 691790886 265871957 985572982 68567863 844583023 523837474 617739128 435045538 926684807 237031902 94857244 223518201 820322239 254077474 666747710 650996229 839864663 574991966 299733310 695509230 950438108 170974538 859410451 746228638 31208815 443029839 904690982 786770321 830753779 760867660 159537587 318367680 167112106 139816182 219754897 343057778 614121512 761712391 98181552 115668882 185943949 324716025 751878065 627468196 177637167 721300568 
9
899956350 78797448 330117932 60396748 809907699 810720092 806706185 410869635 538052633 
34
698722818 417157642 199157714 312928343 484829300 603912683 18258815 605328124 570401555 644692351 79472104 937966391 891281852 440368935 383352164 111273268 749064384 173188904 29452606 454047922 739703648 602621062 487370265 575918341 503587507 277492458 346489676 885551931 913068064 413300607 159055650 329342206 458270626 36076907 
79
965078688 648187695 661999625 650405882 583973139 288534636 151771043 802557478 701400252 241121022 834254998 935672194 73283860 554633828 987795255 551145805 216585518 873168048 50904317 485275047 70342872 972260773 470541153 254210627 168533362 437647393 134647730 330850060 91835363 886313718 329918270 823431351 130548355 238266009 201892841 613395976 771717899 786868044 332892628 548572179 475309528 890948321 811887191 899716148 648394588 539749212 56494342 172626365 728309560 911894091 18978498 28575817 860111533 723614571 124922630 341048371 80823935 216944552 985444762 664819598 922521715 46053145 346866667 939247688 710891538 992531211 266874589 508019974 618311852 681258616 444702217 726302713 827505011 538415542 330211079 554871727 153244508 328169521 446735421 
93
625880315 384535834 912980524 35088374 989101002 784803252 123026877 255753167 148001052 816255928 360290425 580950210 190115983 774711032 649753210 657820196 507763963 473512512 120919107 431729814 195042979 754885549 315602223 29313505 60654559 106569882 129806151 101425969 30581966 435272792 770494416 197305507 289504319 543948823 23385982 577281244 688323245 577930851 790279809 530491033 819710941 182870002 277910015 200499123 891237164 387671703 371755231 558278160 426546478 40571606 925940751 571095242 891273745 34857920 951434780 748501485 66734210 320032235 599210951 439565182 969722469 874175142 330250903 622544608 309445196 980181864 685199840 190437496 92866376 51611716 81599696 976466075 69238771 582531439 574094729 927169519 163507232 495899714 141125075 371147057 380624228 925303948 106581894 180871331 710145234 198908469 865788093 67235487 664688985 371834580 772192594 642059412 904123371 
82
161082889 131657951 129099631 316863223 628432335 533784953 762010799 274388162 470747533 643963150 259169152 628240510 321742737 685433406 230893312 708971498 294115001 755113686 518288748 504163187 709720148 701622238 60599599 910219086 297261726 304096764 643563000 544661204 513488246 948402613 444562627 148498389 724060341 847452859 406729346 769023466 44268789 892894270 114038894 159444477 358226853 657546952 70082430 245465303 561537260 451375047 482640792 689404095 75775127 414817904 294199740 920859796 343680049 414968271 80931001 314529649 307253941 460409579 28963519 351382555 702912737 71927635 510168989 494182700 542345951 107485260 549890806 859043723 999961877 205509083 892665309 390724399 747803666 315935067 248072463 603143930 95919788 911486130 493847711 143459158 149910469 535312290 
87
922377805 349057935 13550556 117393277 919527538 758226620 166371965 646890224 732515816 614459485 712919399 656853038 64195623 198860397 231853476 437329327 96683525 435640109 572326634 543587333 855145528 107270136 855602903 866227247 341147706 146103933 939628833 177790163 527646366 746930928 767948334 937721352 996768319 289618453 957614910 997428165 280321143 102135642 934447757 518138410 151703539 496063228 829237987 689340873 508039868 54396084 502123076 951600441 479343143 509864518 143587526 397744990 794774725 231985249 730003285 352536040 27233945 222468982 862989373 463096864 134678221 648953115 457406716 525970011 913688333 381907300 315182607 521031047 900936080 101227301 728084854 447908268 817408800 991909317 656297556 440146194 336384089 66295046 948560271 160813053 171457182 413817367 878204047 66236364 140094570 905519107 256938772 
39
560387223 791479747 469635325 582345793 17558872 446165157 638410520 280311925 706803317 248625775 899309286 86408643 978384936 206679338 770472755 193052408 344001990 624529529 33848936 890859638 920559263 132200238 35720303 163510724 746767365 922496193 925772280 250903475 596704968 697648262 585746914 63632465 78383571 807141150 224027851 801632742 415384570 723297785 299438175 
47
265349647 486006830 395733365 331619666 39649249 643052245 870867834 51729601 94119461 636650395 653346389 250491092 998839512 416985844 164229740 948264404 195108172 64789792 450958353 546252098 915693274 415327447 740551097 411801710 475236177 42048344 516386107 840869767 250133574 11684196 273627010 724816869 398125335 213180552 307908622 654578429 695235391 305755751 114936184 341696139 566176338 580875494 350610946 35143817 165880740 795311924 816202297 
81
446900119 222668715 631958014 398727941 978321248 255706325 862318308 601514478 217726851 410912500 391217502 764700749 728226968 788305994 628460651 593020794 27479974 821235084 476345012 599899570 44505916 291311343 131245434 989717009 259584302 214567345 403545200 971232688 560824601 830868403 319241463 670757631 101402418 497509826 487729524 915475144 163057009 77988844 846605111 541209281 905972829 23694137 594445470 648273652 784526854 415583244 375951081 485103441 651814459 179271906 781328158 88197480 797654482 773058399 884475728 317462126 883383211 26071765 599396785 376632928 860194228 263375537 606299539 749077639 123440531 866493858 171697097 400539777 387234805 248510102 640158099 72617661 950766892 35862144 376909082 634011693 245403507 829089543 138673776 314428121 810772477 
74
470954420 224681632 54160593 222252351 570154155 630793126 687024448 650180457 363213411 407897452 997310132 857384082 981464845 41834208 146535015 992054025 31895726 252137068 718743345 276260909 391675167 433056401 782814760 362861557 80743166 217479768 146758298 61548293 199934459 280575555 450782265 430220696 694075831 118843411 787208538 98132960 600448538 97097797 898780926 914342468 166139435 277575181 705629577 694409731 168035400 603239778 828527435 383293007 311572971 570193891 40920659 60843271 697346319 702998582 337368257 639179449 175597065 139762673 858641305 922418812 407385695 47934259 369903990 570869824 157108064 910019713 498014374 256450472 833953207 203826022 957639462 948355357 543659099 593872414 
96
604909006 257415212 455697997 234676480 598031007 125331210 364051124 421096157 624456323 626333921 809265880 757120802 208368757 658366788 668328481 880428306 442371578 517817791 503172744 131844750 110703502 227333132 207688326 100185873 344643048 315125122 231215345 217537005 401053694 815068594 999080857 902007172 909170310 738344207 89204663 513341466 339944370 294711087 947138199 528809902 234401858 971136919 55985625 743454551 188990327 177950931 97006731 241476573 927659833 81055912 965767825 251873567 304566836 639604723 533128451 384048200 290683412 127557994 915911304 413137653 633076801 44851356 54482182 69781130 171495355 258641577 79634853 523632953 793956470 160935990 84277509 842183458 209875503 725058587 272744300 699855054 980405145 657092542 413389925 17806701 688169538 70854822 52605680 695946291 309391422 468778264 142910303 689752521 284867323 434446444 463490634 675112703 830226468 711505888 580208918 453919893 
4
688956554 454491030 273663038 798125843 
78
746610537 931681538 559411982 360218242 369629707 374325810 58541153 87363933 893401703 681172098 813001832 688172256 961700581 747254556 857132944 351970232 569018059 586525797 569859651 180976067 533282940 389202010 632042302 484520357 727886004 917847224 407337595 904741803 622993824 981358267 627573355 398702045 125382157 668294694 64947816 219942906 192087038 613889083 160267703 958261327 474631969 832920322 616876017 912416524 473337311 114104764 156793417 85929604 638762455 124491196 708391042 113417500 641035942 613943199 961584063 619848751 306609604 758052065 314796763 758605892 406571965 815844832 866261260 791317514 140594990 521652051 952686944 971742088 115992469 814757799 849146452 219463293 830778172 89301220 317413904 620083553 114905222 710221064 
22
887869762 591704517 28425174 781365033 656614938 585716324 510181435 978384560 564203132 654872485 53028832 538078582 331573802 535699804 738568045 879297296 969150921 130545223 487314300 171955622 700083223 994948302 
43
218050311 514936414 905354473 251264098 265222393 237074959 107281263 259745927 394372124 890710374 613214662 276317921 35535329 953016890 262102769 589849687 191350062 146659617 325257392 475725367 678849813 388467081 537888751 599279436 157439117 271853901 726588163 709613773 776212761 838651086 556736622 192032871 965786023 20817923 372491975 90814013 908778825 412222739 65246183 814232546 966209491 900889097 22893013 
