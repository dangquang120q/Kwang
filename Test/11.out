-1638028815
-2022520139
-1377011919
-1394790065
-1306111860
-1503710008
-1384501600
-1435546173
-1380886970
-1381760580
-1422732078
-1330861094
-1332930916
-1387024573
-1363270093
-1462752075
-1380354895
-1570519840
-1499413552
-1611765018
-1328538745
-1455902769
-1404377946
-1774383155
-1552650989
-1329196068
-1349218355
-1425743010
-1308861495
-1468313887
-1432170106
-1355696351
-1324138294
-1344344375
1941573427
-1380166360
-1352483513
-1409954892
